NitroxServer/ConsoleCommands/Processor/ConsoleCommandProcessor.cs

```diff
@@ -9,54 +9,61 @@ namespace NitroxServer.ConsoleCommands.Processor
 {
     public class ConsoleCommandProcessor
     {
-        private static Dictionary<string, Command> commands = new Dictionary<string, Command>();
-
-        public static void RegisterCommands()
-        {
-            IEnumerable<Command> discoveredCommands = Assembly.GetExecutingAssembly()
-                .GetTypes()
-                .Where(p => typeof(Command).IsAssignableFrom(p) &&
-                            p.IsClass && !p.IsAbstract
-                      )
-                .Select(Activator.CreateInstance)
-                .Cast<Command>();
-
-            foreach (Command cmd in discoveredCommands)
-            {
-                RegisterCommand(cmd);
-            }
-        }
-
-        public static void RegisterCommand(Command command)
-        {
-            commands.Add(command.Name, command);
-        }
-
         public static void ProcessCommand(string msg)
         {
             if (string.IsNullOrWhiteSpace(msg))
             {
                 return;
             }
 
-            Command cmd;
-
             string[] parts = msg.Split()
                                 .Where(arg => !string.IsNullOrEmpty(arg))
                                 .ToArray();
 
-            if (commands.TryGetValue(parts[0], out cmd))
-            {
-                string[] args = parts.Skip(1).ToArray();
+            IEnumerable<Command> discoveredCommands = FindCommands();
 
-                if (cmd.VerifyArgs(args))
+            foreach (Command command in discoveredCommands)
+            {
+                if (command.Name == parts[0])
                 {
-                    cmd.RunCommand(args);
-                    return;
+                    RunCommand(command, parts);
                 }
+                else
+                {
+                    if (command.Alias != null)
+                    {
+                        int index = Array.IndexOf(command.Alias, parts[0]);
+                        if (index != -1)
+                        {
+                            RunCommand(command, parts);
+                        }
+                    }
+                }
+            }
+        }
 
-                Log.Info("Command Invalid: {0}", cmd.Args);
+        private static IEnumerable<Command> FindCommands()
+        {
+            return Assembly.GetExecutingAssembly()
```

Comment:
Those Commands should be passed through the constructor now. If you add an `IEnumerable<Command>` to `ConsoleCommandProcessor` constructor you then don't have to refind the commands on every ProcessCommand and use AutoFac.